### ISSUE TITLE
Can we have an API reference?

### ISSUE THREAD

**Issue by TMiguelT** [spoike/refluxjs#291]
As well as the general readme, I think reflux should have an API.md file that would list the properties of the definition objects that you pass into createStore, createAction etc. as well as their methods. 

At the moment, even `Store#getInitialState` isn't even mentioned in the readme, which is very confusing.

**Comment by srph** [comment]
Good point.

Also, anyone think Reflux needs an improved documentation :sparkles:? Besides splitting the `README` documentation to smaller files, maybe use [RTD](https://readthedocs.org)? imo, a long documentation page with different subjects isn't appropriate to read and edit :-1:.

**Comment by tmbtech** [comment]
@TMiguelT if you want to start the API.md, I'll be happy to contribute.

**Comment by spoike** [comment]
Yes, please. I'm currently limited with other obligations, so any help with getting API documentation through md files up would be helpful.

**Comment by srph** [comment]
Anybody willing to setup a gh-page for a doc site?

**Comment by spoike** [comment]
Sorry for not replying. We've moved the projects to the reflux org, and extracted the core methods to it's own project. We'll be working with setting this up with a main site as well as gh-pages. Probably using hexo.io as static site builder, to bundle in with jsdoc or esdoc.

**Comment by devinivy** [comment]
The reflux-core project has an API reference [here](https://github.com/reflux/reflux-core/tree/master/docs/api).  We'll be trying to improve the presentation and usability of the docs by creating a site some time in the future.  We'll also take PRs to improve the existing docs.  We'd love help, so check out #447 :)
